pyrasterframes/src/main/python/tests/PyRasterFramesTests.py

```diff
@@ -574,7 +574,7 @@ def test_matmul(self):
         nd = r1.cell_type.no_data_value()
         e1 = Tile(np.ma.masked_equal(np.array([[nd, 10],
                                                [nd, nd]], dtype=r1.cell_type.to_numpy_dtype()), nd))
-        self.assertTrue(r1 == e1)
+        self.assertEqual(r1, e1)
```

Comment:
@vpipkt 

I haven't figured out why yet, but `test_matmul` is failing:

```
>       self.assertEqual(r1, e1)
E       AssertionError: Tile([18 chars]ta=[[ 7, 10],
E               [15, 22]],
E         mask=False,[41 chars]9999) != Tile([18 chars]ta=[[--, 10],
E               [--, --]],
E         mask=[[ Tru[72 chars]ud63)

tests/PyRasterFramesTests.py:577: AssertionError
```

Might have happened after I merged `develop` onto `feature/pip-installable`.

Reply:
Likely something to do with this change?

```diff
 class Tile(object):
-    def __init__(self, cells, cell_type):
-        self.cell_type = cell_type
-        self.cells = cells.astype(cell_type.to_numpy_dtype())
+    def __init__(self, cells, cell_type=None):
+        if cell_type is None:
+            # infer cell type from the cells dtype and whether or not it is masked
+            ct = CellType.from_numpy_dtype(cells.dtype)
+            if isinstance(cells, np.ma.MaskedArray):
+                ct = ct.with_no_data_value(cells.fill_value)
+            self.cell_type = ct
+        else:
+            self.cell_type = cell_type
+        self.cells = cells.astype(self.cell_type.to_numpy_dtype())
 
         if self.cell_type.has_no_data():
             nd_value = self.cell_type.no_data_value()
             if np.isnan(nd_value):
                 self.cells = np.ma.masked_invalid(self.cells)
             else:
                 # if the value in the array is `nd_value`, it is masked as nodata
                 self.cells = np.ma.masked_equal(self.cells, nd_value)
 
     def __eq__(self, other):
         if type(other) is type(self):
-            return self.cell_type == other.cell_type and np.array_equal(self.cells, other.cells)
+            return self.cell_type == other.cell_type and np.ma.allequal(self.cells, other.cells)
         else:
             return False
```

cf: https://github.com/locationtech/rasterframes/commit/d29634ac8091560e58afad3629c8f4c59af56d5b

Reply:
I will look into it.

Reply:
Hmm it is passing for me on my local environment against python 3.6. I cleared up a little problem with python 2.7 and now see on circle CI it's passing. 🤷‍♂

Reply:
It's failing for me on my machine.... difference in numpy or similar?

Reply:
hmm yes unfortunately so, it could be a version issue?

I am testing with python 2.7.15 + numpy 1.16.4 OR python 3.6.7 and numpy 1.16.2

Reply:
@metasim can you let me know which versions you're using locally? maybe with a `pip freeze` or `conda list`?

Reply:
Creating issue

Reply:
#156